docs/source/jit_language_reference_statements.rst

```diff
@@ -0,0 +1,389 @@
+
+The ``pass`` statement:
+^^^^^^^^^^^^^^^^^^^^^^
+::
+    pass_stmt ::= "pass"
+
+The ``print`` statement:
+^^^^^^^^^^^^^^^^^^^^^^^^
+::
+    print_stmt ::= "print"
+
+The ``break`` statement:
+^^^^^^^^^^^^^^^^^^^^^^^^
+::
+    break_stmt ::= "break"
+
+The ``continue`` statement:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+::
+    continue_stmt ::= "continue"
+
+The ``import`` statement:
+^^^^^^^^^^^^^^^^^^^^^^^^^
+::
+    import_stmt     ::=  "import" module ["as" identifier] ("," module ["as" identifier])*
+                     | "from" relative_module "import" identifier ["as" identifier]
+                     ("," identifier ["as" identifier])*
+                     | "from" relative_module "import" "(" identifier ["as" identifier]
+                     ("," identifier ["as" identifier])* [","] ")"
+                     | "from" module "import" "*"
+    module          ::=  (identifier ".")* identifier
+    relative_module ::=  "."* module | "."+
+    `identifier_list <https://docs.python.org/3/reference/lexical_analysis.html#grammar-token-identifier>`
+
+The ``future`` statement:
+^^^^^^^^^^^^^^^^^^^^^^^^^
+::
+    future_stmt ::=  "from" "__future__" "import" feature ["as" identifier]
+                 ("," feature ["as" identifier])*
+                 | "from" "__future__" "import" "(" feature ["as" identifier]
+                 ("," feature ["as" identifier])* [","] ")"
+    feature     ::=  identifier
+    `identifier <https://docs.python.org/3/reference/lexical_analysis.html#grammar-token-identifier>`
+
+
+Compound Statements
+~~~~~~~~~~~~~~~~~~~
+
+The following section describes the syntax of Compound statements that are supported in TorchScript.
+The section also highlights how Torchscript differs from regular Python statements.
+It is modeled after `the compound statements chapter of the Python language reference <https://docs.python.org/3/reference/compound_stmts.html>`_.
+
+The ``if`` statement:
+^^^^^^^^^^^^^^^^^^^^^
+* Torchscript supports both basic ``if/else`` and ternary ``if/else``.
+
+Basic ``if/else`` statement:
+""""""""""""""""""""""""""""
+
+::
+    if_stmt ::=  "if" assignment_expression ":" suite
+             ("elif" assignment_expression ":" suite)
+             ["else" ":" suite]
+    `assignment_expression <https://docs.python.org/3/reference/expressions.html#grammar-token-assignment-expression>`
+    `suite <https://docs.python.org/3/reference/compound_stmts.html#grammar-token-suite>`
+
+Ternary ``if/else`` statement:
+""""""""""""""""""""""""""""""
+
+::
+    if_stmt ::=  return [expression_list] "if" assignment_expression "else" [expression_list]
+    `assignment_expression <https://docs.python.org/3/reference/expressions.html#grammar-token-assignment-expression>`
+    `expression_list <https://docs.python.org/3/reference/expressions.html#grammar-token-expression-list>`
+
+* ``Tensor`` with 1 dimension is promoted to ``bool``
```

Comment:
The rule for this promotion isn't clear to me. Should we go into more details like how are undefined tensors are promoted?

Reply:
How can we create an undefined tensor. I tried out torch.Tensor(), but that didn't get through compilation.